libs/auth/src/cast_vote_record_hashes.ts

```diff
@@ -78,29 +78,25 @@ export function readableFileFromDisk(filePath: string): ReadableFile {
 
 /**
  * Computes a hash for a single cast vote record, incorporating the cast vote record's directory
- * name, file names, and file contents. The directory summary that we hash specifically mirrors the
- * output of:
- * find <directoryName> -type f | sort | xargs sha256sum
+ * name, report file name, and report contents. The summary that we hash specifically mirrors
+ * the output of:
+ * ```
+ * sha256sum <cvrDirectoryName>/<CastVoteRecordExportFileName.CAST_VOTE_RECORD_REPORT>
+ * ```
  */
-export async function computeSingleCastVoteRecordHash({
-  directoryName,
-  files,
-}: {
-  directoryName: string;
-  files: HashableFile[];
-}): Promise<string> {
-  const filesSorted = [...files].sort((file1, file2) =>
-    file1.fileName.localeCompare(file2.fileName)
+export async function computeSingleCastVoteRecordHash(
+  cvrDirectoryName: string,
+  cvrReportFile: HashableFile
+): Promise<string> {
+  // Ensure that a directory path wasn't accidentally provided instead of a directory name
+  assert(!cvrDirectoryName.includes('/'));
+
+  const cvrReportPath = path.join(
+    cvrDirectoryName,
+    CastVoteRecordExportFileName.CAST_VOTE_RECORD_REPORT
   );
-  const hasher = sha256.create();
-  for (const file of filesSorted) {
-    const filePath = path.join(directoryName, file.fileName);
-    // Be extra cautious and prevent spoofing the directory summary by using directory/file names
-    // with newlines
-    assert(!filePath.includes('\n'));
-    hasher.update(`${await file.computeSha256Hash()}  ${filePath}\n`);
-  }
-  return hasher.hex();
+  const cvrDirectorySummary = `${await cvrReportFile.computeSha256Hash()}  ${cvrReportPath}\n`;
+  return sha256(cvrDirectorySummary);
```

Comment:
You might be wondering why we have to hash this special string and can't just hash the `cast-vote-record-report.json`, alone, as the hash for a single CVR. Incorporating the name of the CVR directory into the hash (by way of `cvrReportPath`) is important because it prevents renaming a CVR directory.

Without this, CVR directory renames that keep the overall alphabetical order of the CVR directories the same could go undetected.

It was actually a `libs/auth` test that reminded me to cover this case! It failed on my first pass at this update, where I was directly using the hash of the `cast-vote-record-report.json` as the hash for the CVR.

https://github.com/votingworks/vxsuite/blob/93b214c8dc9ca8eca583e118650f919e8f662bc2/libs/auth/src/artifact_authentication.test.ts#L314-L315